accounts-db/src/accounts_db.rs

```diff
@@ -6375,6 +6375,9 @@ impl AccountsDb {
                     .map(|should_flush_f| should_flush_f(key, account))
                     .unwrap_or(true);
                 if should_flush {
+                    // Add `key` to uncleaned_pubkeys since we know this account will now be
+                    // written to a storage, and should be visited by `clean`.
+                    self.uncleaned_pubkeys.entry(slot).or_default().push(*key);
```

Comment:
I think I'd prefer to use `accounts` that is collected here after `filter_map` and lookup the `slot` entry in `uncleaned_pubkeys` once. So, after collecting, iter the collectd `accounts` and push or append them all. This reduces locks and runtime of adding these things to the slot entry in `uncleaned_pubkeys` I imagine.

Reply:
great. ty.